From dd61a944f0c147296956918fd687c02cd294475d Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 10 Mar 2022 10:01:09 -0300
Subject: [PATCH 031/142] iconvdata: Suppress clang warning on locale
 definitions

clang issues an warning adding '{unsigned} int' to a string does not
append to the string, however it is exactly what code means here.
---
 iconvdata/gbk.c             | 5 +++++
 iconvdata/iso-2022-cn-ext.c | 5 +++++
 iconvdata/iso-2022-cn.c     | 5 +++++
 3 files changed, 15 insertions(+)

diff --git a/iconvdata/gbk.c b/iconvdata/gbk.c
index d1365ca188..59bfda1b24 100644
--- a/iconvdata/gbk.c
+++ b/iconvdata/gbk.c
@@ -13212,6 +13212,10 @@ static const char __gbk_from_ucs4_tab12[][2] =
   }
 #include <iconv/loop.c>
 
+/* clang issues an warning adding 'unsigned int' to a string does not append
+   to the string, however it is exactly what code means here.  */
+DIAG_PUSH_NEEDS_COMMENT_CLANG;
+DIAG_IGNORE_NEEDS_COMMENT_CLANG (13, "-Wstring-plus-int");
 
 /* Next, define the other direction.  */
 #define MIN_NEEDED_INPUT	MIN_NEEDED_TO
@@ -13493,6 +13497,7 @@ static const char __gbk_from_ucs4_tab12[][2] =
 #define LOOP_NEED_FLAGS
 #include <iconv/loop.c>
 
+DIAG_POP_NEEDS_COMMENT_CLANG;
 
 /* Now define the toplevel functions.  */
 #include <iconv/skeleton.c>
diff --git a/iconvdata/iso-2022-cn-ext.c b/iconvdata/iso-2022-cn-ext.c
index 0be9278740..942a5e0542 100644
--- a/iconvdata/iso-2022-cn-ext.c
+++ b/iconvdata/iso-2022-cn-ext.c
@@ -387,6 +387,10 @@ enum
 #define LOOP_NEED_FLAGS
 #include <iconv/loop.c>
 
+/* clang issues an warning adding 'int' to a string does not append
+   to the string, however it is exactly what code means here.  */
+DIAG_PUSH_NEEDS_COMMENT_CLANG;
+DIAG_IGNORE_NEEDS_COMMENT_CLANG (13, "-Wstring-plus-int");
 
 /* Next, define the other direction.  */
 #define MIN_NEEDED_INPUT	TO_LOOP_MIN_NEEDED_FROM
@@ -669,6 +673,7 @@ DIAG_POP_NEEDS_COMMENT;
 #define LOOP_NEED_FLAGS
 #include <iconv/loop.c>
 
+DIAG_POP_NEEDS_COMMENT_CLANG;
 
 /* Now define the toplevel functions.  */
 #include <iconv/skeleton.c>
diff --git a/iconvdata/iso-2022-cn.c b/iconvdata/iso-2022-cn.c
index 0990595e96..2f0fb5987d 100644
--- a/iconvdata/iso-2022-cn.c
+++ b/iconvdata/iso-2022-cn.c
@@ -227,6 +227,10 @@ enum
 #define UPDATE_PARAMS		*setp = set | ann
 #include <iconv/loop.c>
 
+/* clang issues an warning adding 'int' to a string does not append
+   to the string, however it is exactly what code means here.  */
+DIAG_PUSH_NEEDS_COMMENT_CLANG;
+DIAG_IGNORE_NEEDS_COMMENT_CLANG (13, "-Wstring-plus-int");
 
 /* Next, define the other direction.  */
 #define MIN_NEEDED_INPUT	TO_LOOP_MIN_NEEDED_FROM
@@ -401,6 +405,7 @@ enum
 #define UPDATE_PARAMS		*setp = set | ann
 #include <iconv/loop.c>
 
+DIAG_POP_NEEDS_COMMENT_CLANG;
 
 /* Now define the toplevel functions.  */
 #include <iconv/skeleton.c>
-- 
2.34.1

